.github/workflows/build.yml

```diff
@@ -0,0 +1,85 @@
+name: Build
+
+on:
+  push:
+    tags:
+      - "v*"
+    branches:
+      - "main"
+  pull_request:
+env:
+  CARGO_TERM_COLOR: always
+  RUSTFLAGS: -Dwarnings
+  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
+
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    name: Check
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        run: rustup install stable
+      - name: Check Formatting
+        run: cargo fmt --check
+      - name: Lint
+        run: cargo clippy --all-targets --all-features --release
```

Comment:
I would consider even adding `-- -D clippy::pedantic -D clippy::unwrap_used` at the end there (I think that's how the command is meant to be structured?)

I want to encourage only using `expect` instead of unwrap in the final code.

Reply:
Done in `main.rs`